Source/AwsCommonRuntimeKit/crt/Utilities.swift

```diff
@@ -90,6 +90,16 @@ func withOptionalAWSByteCursorFromData<Result>(
     }
 }
 
+func withOptionalByteCursorPointerFromData<Result>(
+    to data: Data?, _ body: (UnsafePointer<aws_byte_cursor>?) throws -> Result) rethrows -> Result {
+    guard let _data = data else {
+        return try body(nil)
+    }
```

Comment:
Debatable: I feel more safe to have a different name for a different variable type to avoid confusion. If it is not a strong against, I'd prefer keep it like this.

Reply:
Plus, I think it's better to shadow because you are not going to use data after the unwrap, so it should be unwrapped throughout the function, and it is not ideal to have both an optional and an unwrapped value in the scope if only one is needed and used.